src/platform/plugins/shared/discover/public/application/main/state_management/discover_internal_state_container.ts

```diff
@@ -126,6 +130,22 @@ export function getInternalStateContainer() {
         ...prevState,
         adHocDataViews: newAdHocDataViewList,
       }),
+      setDefaultProfileAdHocDataViews:
+        (prevState: InternalState) => (defaultProfileAdHocDataViews: DataView[]) => {
+          const adHocDataViews = prevState.adHocDataViews
+            .filter((dataView) => !prevState.defaultProfileAdHocDataViewIds.includes(dataView.id!))
+            .concat(defaultProfileAdHocDataViews);
+
+          const defaultProfileAdHocDataViewIds = defaultProfileAdHocDataViews.map(
+            (dataView) => dataView.id!
+          );
+
+          return {
+            ...prevState,
+            adHocDataViews,
```

Comment:
Maybe it's a separate issue but ad hoc data views don't appear in the flyout when creating O11y alert from Discover page.